src/components/App.js

```diff
@@ -1,39 +1,55 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
-import { checkURIforError, getTokenFromURI, getAuthURL } from '../config/authConfig';
-import macaroon from '../assets/img/Macaroonicon.png';
+import { checkURIforError, getTokenFromURI, getTokenFromLocal } from '../config/authConfig';
+import { Dashboard } from './Dashboard';
+import { Sidebar } from './Sidebar';
+import { Welcome } from './Welcome';
+import { LoginLink } from './LoginLink';
+import UserPlaylists from './UserPlaylists';
+import CurrentPlaylist from './CurrentPlaylist';
 
 
 class App extends React.Component {
   state = {
     token: '',
-    user: '',
-    error: false,
-    errorMessage: '',
+    user: {
+      name: '',
+      id: '',
+      error: false,
+      errorMessage: '',
+    },
+    currentPlaylist: {
+      name: '',
+    },
   }
 
   componentDidMount() {
+    let token = getTokenFromURI();
     const authError = checkURIforError();
-    const token = getTokenFromURI();
+    if (!token) {
+      token = getTokenFromLocal();
+    }
     this.setAuthError(authError);
     if (token) {
-      this.setToken(token);
-      this.getProfile(token);
+      this.setState({ token });
+      this.fetchProfile(token);
     }
   }
 
-  setToken = (token) => {
-    this.setState({ token });
-  }
 
   setAuthError = (error) => {
     const errorMessage = error
-      ? 'Access denied by user'
+      ? 'Profile access revoked by user.'
       : '';
-    this.setState({ error, errorMessage });
+    this.setState({
+      user: {
+        error,
+        errorMessage,
+      },
+    });
   }
 
-  getProfile = (token) => {
+  fetchProfile = (token) => {

```

Comment:
Try refactoring this using async/await. I'd also pull this network logic out into it's own function in a different file, so you can invoke the function and have it return the value you want in a single line, rather than managing the network request inside the component.

Reply:
Glad you think so because I just got started on that yesterday